### PR TITLE
pass ssh_password and some options provided by Net::SSH.configuration_for to session

### PR DESCRIPTION
**Pull request by sawanoboly**
option `--ssh-password` will be ignored by ssh session.

----

- I'll add spec for session_options
- And, I think that we should not truncate options from ssh_config.

### CODE REVIEW

**Comment by sawanoboly** [comment]
removed WIP :sunglasses: 

Please review, @coderanger

File: lib/chef/knife/ssh.rb [hunk 1]
```diff
@@ -219,23 +219,28 @@ def search_nodes
       # @return [Hash<Symbol, Object>]
       def session_options(host, port, user=nil)
         ssh_config = Net::SSH.configuration_for(host)
-        {}.tap do |opts|
+        ssh_config.tap do |opts|
```
Comment (coderanger) [inline lib/chef/knife/ssh.rb]: Why this change?

Comment (sawanoboly) [inline lib/chef/knife/ssh.rb]: `Net::SSH.configuration_for` returns several useful options, such as compression_level, timeout and etc.

But, session_options discards them.

Comment (coderanger) [inline lib/chef/knife/ssh.rb]: Yes, that is intentional to maintain compat with how this code used to work.

Comment (sawanoboly) [inline lib/chef/knife/ssh.rb]: I think that using all options is okay except send_env.

Comment (coderanger) [inline lib/chef/knife/ssh.rb]: That is a much bigger change than simply adding the password argument. /ping @chef/client-core

Comment (sawanoboly) [inline lib/chef/knife/ssh.rb]: For example, in exist spec.

https://github.com/chef/chef/blob/master/spec/unit/knife/ssh_spec.rb#L189-L194

`:timeout` is set, but it'll be ignore. isn't it?

Comment (sawanoboly) [inline lib/chef/knife/ssh.rb]: I'm writing a spec after I updated of the password, it was worrisome.

**Comment by sawanoboly** [comment]
@coderanger 
Shall I separate https://github.com/higanworks/chef/commit/e5a78b50028be99744c7c674efe694a373b868ef as another PR?

It's just recover regression.

**Comment by thommay** [comment]
@sawanoboly do you want to rebase this patch and clean it up now that the ssh_password bit has been merged?

**Comment by sawanoboly** [comment]
@thommay 
Yes, I'll rebase my branch and create as new PR.
